torch/csrc/distributed/autograd/engine/dist_engine.cpp

```diff
@@ -180,6 +218,22 @@ void DistEngine::computeDependencies(
     // Create a dummy GraphRoot and run init_to_execute with it.
     GraphRoot dummyRoot(edges, {});
     graphTask->init_to_execute(dummyRoot, outputEdges);
+    for (auto& pr : graphTask->exec_info_) {
+      auto fn = pr.first;
+      auto& execInfo = pr.second;
+      if (!execInfo.captures_) {
+        continue;
+      }
+      if (auto accumulateGradFn = dynamic_cast<AccumulateGrad*>(fn)) {
+        for (auto& capture : *execInfo.captures_) {
+          capture.hooks_.push_back(
+              std::make_unique<DistAccumulateGradCapturePreHook>(
+                  std::dynamic_pointer_cast<AccumulateGrad>(
+                      accumulateGradFn->shared_from_this()),
+                  autogradContext));
+        }
+      }
```

Comment:
We used to have this logic in the BFS above (line 166 - 185) to avoid traversing exec_info again. Is there a reason we moved this logic here instead?

Reply:
Yes, in the BFS the `exec_info` doesn't even have entries for `AccumulateGrad` nodes.  The call of `init_to_execute()` after the BFS adds those entries. I don't want to replicate the similar logic here.